hbase/src/main/java/com/yahoo/ycsb/db/HBaseClient10.java

```diff
@@ -246,17 +266,20 @@ public int read(String table, String key, Set<String> fields, HashMap<String,Byt
             //do nothing for now...need to understand HBase concurrency model better
             return ServerError;
         }
-
-        if (r.isEmpty()) {
+        if (multiGetsCount == 0) {
+          if (r[0].isEmpty()) {
             return NoMatchingRecord;
+          }
         }
-        for (Cell c : r.listCells()) {
+        for (Result res : r) {
```

Comment:
If multiGetBatchSize > 0, only then we will use this whole multiGet based option. Else we will be going with the usual way of issuing a get and the result is stored in r[0]. We have this code anyway
          if (multiGetBatchSize == 0) {
          if (r[0].isEmpty()) {
            return NoMatchingRecord;
          }
        }
As you said if one of the get does return an empty result we need to have this
        for (Result res : r) {
          if (!res.isEmpty()) {